packages/yoastseo/GLOSSARY.md

```diff
@@ -0,0 +1,184 @@
+# YoastSEO.js Glossary
+
+A comprehensive glossary of terms and concepts used in YoastSEO.js.
+
+## Core Concepts
+
+### <a name="paper"></a>Paper
+A value object that encapsulates all content and metadata to be analyzed. The Paper object is immutable and serves as the primary input for all analyses.
+
+**Properties include:**
+- text: The main content to analyze
+- title: The SEO title or page title
+- keyword: The focus keyphrase
+- description: Meta description
+- url: The slug/URL of the page
+- locale: Language code (e.g., en_US)
+- permalink: Full URL
+
+**Example:**
+```javascript
+const paper = new Paper("This is the main content", {
+    title: "Example Title",
+    keyword: "example",
+    description: "This is a meta description",
+    url: "example-page",
+    locale: "en_US"
+});
+```
+
+### <a name="assessment"></a>Assessment
+A single analysis unit that evaluates one specific aspect of content. Each assessment:
+- Has a specific purpose (e.g., keyword density analysis)
+- Produces a score (0-100)
+- Can generate improvement suggestions
+- May include text markers for visual feedback
+
+**Example Assessment Structure:**
+```javascript
+class KeywordDensityAssessment extends Assessment {
+    getResult(paper, researcher) {
+        const density = researcher.getResearch("keywordDensity");
+        return {
+            score: this.calculateScore(density),
+            text: this.translateScore(density)
+        };
+    }
+}
+```
+
+### <a name="assessor"></a>Assessor
+A collection manager that:
+- Coordinates multiple assessments
+- Determines which assessments to run
+- Aggregates assessment results
+- Calculates overall scores
+
+Types of assessors include:
+- SEOAssessor: Focuses on search engine optimization
+- ReadabilityAssessor: Analyzes text readability
+- CornerStoneAssessor: Applies stricter rules for important content
+
+```mermaid
+graph TD
+    A[Assessor] --> B[SEOAssessor]
+    A --> C[ReadabilityAssessor]
+    A --> D[CornerStoneAssessor]
+    B --> E[KeywordAssessment]
+    B --> F[URLAssessment]
+    C --> G[SentenceLengthAssessment]
+    C --> H[ParagraphLengthAssessment]
+```
+
+### <a name="researcher"></a>Researcher
+The research component that:
+- Performs linguistic analysis
+- Caches research results
+- Provides research data to assessments
+
+Common research types:
+- Word count
+- Sentence detection
+- Keyword presence
+- Morphological analysis
+
+**Example Research Usage:**
+```javascript
+const researcher = new Researcher(paper);
+const wordCount = researcher.getResearch("wordCountInText");
+const sentences = researcher.getResearch("sentences");
+```
+
+### <a name="score"></a>Score
+A numeric representation (0-100) of content quality for a specific aspect.
+
+Score ranges:
+- 0-40: Bad (Red)
+- 41-70: Needs Improvement (Orange)
+- 71-100: Good (Green)
```

Comment:
```suggestion
- 0-30: Bad (Red)
- 31-60: Needs Improvement (Orange)
- 61-90: Good (Green)
```

Reply:
@mhkuu There seems to be some confusion somewhere around the scoring, and I'm not sure what to rely on.
Here's one of the spec tests:
https://github.com/Yoast/wordpress-seo/blob/5d7ffe3f1b99c156ae606d871c994934bbdd1bd4/packages/yoastseo/spec/scoring/renderers/AssessorPresenterSpec.js#L34-L46
Where do you get the adapted scale from?